### PR TITLE
Crypto Tracker

### PR DESCRIPTION
**Pull request by Ian-Gabaraev**
```
import requests

from plugin import plugin
	
# ANSI escape sequences to print in color
    
    
class color:
    GREEN = '\033[92m'
    RED = '\033[91m'
    TAIL = '\033[0m'

	
# List of default crypto pairs
favorite = [
    'BTC/USD',
    'BTC/ETH',
    'BTC/LTC',
    'BTC/XRP'
]

	
# If the price is down, print in red. If the price is up, print in green
def print_in_color(change):
    if float(change) < 0:
        return color.RED + str(change) + color.TAIL
    else:
        return color.GREEN + str(change) + color.TAIL

	
# Main function. It requires the base and target
# crypto symbols, e.g. [BTC, XRP] 
# to build a URL.


def query(base, target):
	
    # Bulding a url from base and target currency symbols
    # It must look like this: https://api.cryptonator.com/api/ticker/btc-eth
	
    url = 'https://api.cryptonator.com/api/ticker/{}-{}'.format(base.lower(), target.lower())
	
    try:
        response = requests.get(url).json()
        price = response['ticker']['price']
        change = response['ticker']['change']
	
	
    # This error occurs if the pair is non-existent
    except KeyError:
        print "Wrong pair {}/{}! \nFull list of symbols is here: https://coinmarketcap.com/all/views/all/\n".format(base, target)
	
   # Results
    else:
        print "\t{}/{}\nPrice: {}\nChange: {} {}\n".format(base, target, price, print_in_color(change), target)

	
# Iterate over your favorite crypto pairs
@plugin("cryptotracker")
def check_favorites(jarvis, s):
    for pair in favorite:
        base, target = pair.split('/')
        query(base, target)
```

### CODE REVIEW

**Comment by TobiOkanlawon** [comment]
Dear Sukeesh,

I'm kinda new to git and programming as a whole. I wanna work on this 
project. Sorry if I'm communicating on the wrong channel, I'll also need 
someone to set  me right on that too. Your repo (Jarvis) was listed on 
some website as a good place to start contributing to O.S.S. I liked it 
because I had a similar idea. I'll need to know what I need to do to 
start working on the project with you all.

I have already forked and cloned the repo, installed the packages on 
requirements.txt. I just need to know what to do next and all, I took 
interest in the voice control issue.

BTW, why is there a requirements.txt when ./setup.sh does everything on 
the list for you. I actually spent all my time downloading everything on 
that requirements file.

Regards,

Toby

**Comment by pnhofmann** [comment]
@TobiOkanlawon 
Not really the best place, but
> BTW, why is there a requirements.txt when ./setup.sh does everything on the list for you.
Just common practice to have all requirements well-arranged in one place. By the way: readme clearly states, all you need to do is call setup.sh ;). Let's discuss everything else in Voice controll issuse ;).

**Comment by pnhofmann** [comment]
@Ian-Gabaraev 

There's the commit! Just - why are you trying to merge in sukeesh:beforepluginmanager and not master?

**Comment by pnhofmann** [comment]
Something else that I've learned: Wasn't aware PRs base branch can be changed ;).

**Comment by pnhofmann** [comment]
Another thing: PEP8.

https://www.python.org/dev/peps/pep-0008/

Basically every programming language has some rules how to format code (e.g. always indent 4 Spaces, never tab). For Python that's PEP8, which we do follow, because it just makes reading code less difficult.

Result from our checking-tool (that's why it reads "1 failed check"):

```
$ flake8 --select E,W --max-line-length=140 --ignore E722,W503,W504,E128 jarviscli/
jarviscli/plugins/cryptotracker.py:4:1: E265 block comment should start with '# '
jarviscli/plugins/cryptotracker.py:5:1: E302 expected 2 blank lines, found 1
jarviscli/plugins/cryptotracker.py:10:1: E265 block comment should start with '# '
jarviscli/plugins/cryptotracker.py:11:1: E305 expected 2 blank lines after class or function definition, found 1
jarviscli/plugins/cryptotracker.py:12:1: E101 indentation contains mixed spaces and tabs
jarviscli/plugins/cryptotracker.py:12:1: W191 indentation contains tabs
jarviscli/plugins/cryptotracker.py:13:1: W191 indentation contains tabs
jarviscli/plugins/cryptotracker.py:14:1: W191 indentation contains tabs
jarviscli/plugins/cryptotracker.py:15:1: W191 indentation contains tabs
jarviscli/plugins/cryptotracker.py:18:1: E265 block comment should start with '# '
jarviscli/plugins/cryptotracker.py:19:1: E302 expected 2 blank lines, found 1
jarviscli/plugins/cryptotracker.py:20:1: W191 indentation contains tabs
jarviscli/plugins/cryptotracker.py:20:2: E117 over-indented
jarviscli/plugins/cryptotracker.py:21:1: W191 indentation contains tabs
jarviscli/plugins/cryptotracker.py:21:3: E117 over-indented
jarviscli/plugins/cryptotracker.py:22:1: W191 indentation contains tabs
jarviscli/plugins/cryptotracker.py:23:1: W191 indentation contains tabs
jarviscli/plugins/cryptotracker.py:23:3: E117 over-indented
jarviscli/plugins/cryptotracker.py:25:1: E265 block comment should start with '# '
jarviscli/plugins/cryptotracker.py:26:1: E265 block comment should start with '# '
jarviscli/plugins/cryptotracker.py:26:33: W291 trailing whitespace
jarviscli/plugins/cryptotracker.py:27:1: E265 block comment should start with '# '
jarviscli/plugins/cryptotracker.py:29:1: E302 expected 2 blank lines, found 1
jarviscli/plugins/cryptotracker.py:31:1: W191 indentation contains tabs
jarviscli/plugins/cryptotracker.py:31:2: E117 over-indented (comment)
jarviscli/plugins/cryptotracker.py:31:2: E265 block comment should start with '# '
jarviscli/plugins/cryptotracker.py:32:1: W191 indentation contains tabs
jarviscli/plugins/cryptotracker.py:32:2: E117 over-indented (comment)
jarviscli/plugins/cryptotracker.py:32:2: E265 block comment should start with '# '
jarviscli/plugins/cryptotracker.py:34:1: W191 indentation contains tabs
jarviscli/plugins/cryptotracker.py:34:2: E117 over-indented
jarviscli/plugins/cryptotracker.py:36:1: W191 indentation contains tabs
jarviscli/plugins/cryptotracker.py:37:1: W191 indentation contains tabs
jarviscli/plugins/cryptotracker.py:37:3: E117 over-indented
jarviscli/plugins/cryptotracker.py:39:1: W191 indentation contains tabs
jarviscli/plugins/cryptotracker.py:40:1: W191 indentation contains tabs
jarviscli/plugins/cryptotracker.py:43:1: W191 indentation contains tabs
jarviscli/plugins/cryptotracker.py:43:2: E303 too many blank lines (2)
jarviscli/plugins/cryptotracker.py:43:2: E265 block comment should start with '# '
jarviscli/plugins/cryptotracker.py:44:1: W191 indentation contains tabs
jarviscli/plugins/cryptotracker.py:45:1: W191 indentation contains tabs
jarviscli/plugins/cryptotracker.py:45:3: E117 over-indented
jarviscli/plugins/cryptotracker.py:47:1: W191 indentation contains tabs
jarviscli/plugins/cryptotracker.py:47:2: E265 block comment should start with '# '
jarviscli/plugins/cryptotracker.py:48:1: W191 indentation contains tabs
jarviscli/plugins/cryptotracker.py:49:1: W191 indentation contains tabs
jarviscli/plugins/cryptotracker.py:49:3: E117 over-indented
jarviscli/plugins/cryptotracker.py:51:1: E265 block comment should start with '# '
jarviscli/plugins/cryptotracker.py:54:1: W191 indentation contains tabs
jarviscli/plugins/cryptotracker.py:54:2: E117 over-indented
jarviscli/plugins/cryptotracker.py:55:1: W191 indentation contains tabs
jarviscli/plugins/cryptotracker.py:55:3: E117 over-indented
jarviscli/plugins/cryptotracker.py:56:1: W191 indentation contains tabs
```

You can also run this 'flake8 --select E,W --max-line-length=140 --ignore E722,W503,W504,E128 jarviscli/' locally on your cli.

Please fix these formatting errors and I'll merge.

**Comment by Ian-Gabaraev** [comment]
Thanks! I have made corrections.

**Comment by pnhofmann** [comment]
Hmm, still failing... Remember that you can always check with this command:
``flake8 --select E,W --max-line-length=140 --ignore E722,W503,W504,E128 jarviscli/``

Yeah, just remembered: Totally forgot to mention (Sorry): There is in fact a tool to do the work for you ;). Just try:

``autopep8 --in-place --aggressive jarviscli/plugins/cryptotracker.py``

**Comment by Ian-Gabaraev** [comment]
> Hmm, still failing... Remember that you can always check with this command:
> `flake8 --select E,W --max-line-length=140 --ignore E722,W503,W504,E128 jarviscli/`
> 
> Yeah, just remembered: Totally forgot to mention (Sorry): There is in fact a tool to do the work for you ;). Just try:
> 
> `autopep8 --in-place --aggressive jarviscli/plugins/cryptotracker.py`

Thanks for mentioning autopep8 :) It should work now.

File: jarviscli/plugins/cryptotracker.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        base.lower(), target.lower())
+
+    try:
+        response = requests.get(url).json()
+        price = response['ticker']['price']
+        change = response['ticker']['change']
+
+    # This error occurs if the pair is non-existent
+    except KeyError:
+        print "Wrong pair {}/{}! \nFull list of symbols is here: https://coinmarketcap.com/all/views/all/\n".format(base, target)
```
Comment (pnhofmann) [inline jarviscli/plugins/cryptotracker.py]: You are using Python 2? Yeah, that's mean: With Python 3 you have to add ().
```suggestion
        print("Wrong pair {}/{}! \nFull list of symbols is here: https://coinmarketcap.com/all/views/all/\n".format(base, target))
```

File: jarviscli/plugins/cryptotracker.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+   # Results
+    else:
+        print "\t{}/{}\nPrice: {}\nChange: {} {}\n".format(base, target, price, print_in_color(change), target)
```
Comment (pnhofmann) [inline jarviscli/plugins/cryptotracker.py]: Same here:
```suggestion
        print("\t{}/{}\nPrice: {}\nChange: {} {}\n".format(base, target, price, print_in_color(change), target))
```

File: jarviscli/plugins/cryptotracker.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    try:
+        response = requests.get(url).json()
+        price = response['ticker']['price']
+        change = response['ticker']['change']
+
+    # This error occurs if the pair is non-existent
+    except KeyError:
+        print "Wrong pair {}/{}! \nFull list of symbols is here: https://coinmarketcap.com/all/views/all/\n".format(base, target)
+
+   # Results
```
Comment (pnhofmann) [inline jarviscli/plugins/cryptotracker.py]: And here's a space missing
```suggestion
    # Results
```

**Comment by Ian-Gabaraev** [comment]
> You should be able to apply small changes with one click here in github. Actually I just wanted to try out this github-feature ;).

Hello, pnhofmann!
Seems like we are good to go :)

**Comment by pnhofmann** [comment]
Yep, so let's merge ;). Thanks!

**Pull request merged**